### PR TITLE
Relocate the kotlin-stdlib in the Gradle Plugin

### PR DESCRIPTION
**Pull request by martinbonnin**
This is mainly an exercise to see what it takes to relocate the kotlin-stdlib and be able to use latest Kotlin versions regardless of the Gradle version used (see [issue #16345](https://github.com/gradle/gradle/issues/16345)  on the Gradle repo)

This PR decouples the Gradle and Apollo Kotlin versions. That's for the advantages. In terms of drawbacks, it's fragile as there could be edge cases or side effects that weren't caught by tests (but so far so good 🤞). It's also adding ~1min of build time for every change to the compiler or Gradle plugin because of R8 needing to be run again. 

The meat of the PR is the `apollo-gradle-build` build.gradle.kts and rules.pro files. There are also changes in the compiler required to remove reflection from the codegen. For more information, you can also read the [blog post](https://blog.mbonnin.net/use-kotlin-15-in-your-gradle-plugins)

### CODE REVIEW

File: apollo-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -1,114 +1,73 @@
...
 
   testImplementation(groovy.util.Eval.x(project, "x.dep.junit"))
   testImplementation(groovy.util.Eval.x(project, "x.dep.truth"))
   testImplementation(groovy.util.Eval.x(project, "x.dep.assertj"))
   testImplementation(groovy.util.Eval.x(project, "x.dep.okHttp.mockWebServer4"))
 }
 
-val shadowJarTask = tasks.named("shadowJar", com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar::class.java)
-val shadowPrefix = "com.apollographql.relocated"
+if (true) {
```
Comment (BoD) [inline apollo-gradle-plugin/build.gradle.kts:44]: Is this to quickly disable it when working on the plugin?

**Review by BoD — APPROVED** [review]
👏

**Review by BoD — APPROVED** [review]

File: apollo-gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (martinbonnin) [reply apollo-gradle-plugin/build.gradle.kts:44]: Yep, exactly!
